lib/manageiq/api/common/user.rb

```diff
@@ -0,0 +1,105 @@
+module ManageIQ
+  module API
+    module Common
+      class User
+        IDENTITY_KEY = 'x-rh-identity'.freeze
+
+        def self.raw_current=(raw_current)
+          Thread.current[:attr_raw_current] = raw_current
+        end
+
+        def self.raw_current
+          Thread.current[:attr_raw_current]
+        end
+
+        def self.current
+          raise "Current User has not been set" unless Thread.current[:attr_current_user]
+          Thread.current[:attr_current_user]
+        end
+
+        def self.current=(user)
+          raise StandardError, "Must be a ManageIQ::API::Common::User" unless user.is_a?(ManageIQ::API::Common::User)
+          Thread.current[:attr_current_user] = user
+        end
+
+        def self.from_header(raw_header)
+          self.raw_current = raw_header
+          raw_header.nil? ? Thread.current[:attr_current_user] = nil : self.current = new(raw_current)
```

Comment:
from_header shouldn't set the current, or at least to a casual reader they would never expect that side effect.  I would expect a method named from_header to essentially just be a factory method that returns a User object form a header.

That being said, I don't think this is needed as all based on my other comment.